src/pyhf/infer/__init__.py

```diff
@@ -78,30 +74,35 @@ def hypotest(
     par_bounds = par_bounds or pdf.config.suggested_bounds()
     tensorlib, _ = get_backend()
 
-    asimov_mu = 0.0
-    asimov_data = generate_asimov_data(asimov_mu, data, pdf, init_pars, par_bounds)
+    calc = AsymptoticCalculator(data, pdf, init_pars, par_bounds, qtilde=qtilde)
```

Comment:
can we not cache the calculator in some way?

Reply:
I don't think the construction is very expensive